S49P19 #32 Admin Server Repo Checks: Service Integration

Itxaka opened the pull request:
- Acomodate the code to the actual API response
- Create a static fake api call to test the admin repocheck page
- Acomodate the tests to the changes
- Sync test data with API response
- Change the repo keys to be in sync with how we use them on the node repo checks
- Acomodate the controller to deal with the API returning a products view rather than a per-repo response.

Commit:
- Acomodate the code to the actual API response
 - Create a static fake api call to test the admin repocheck page
 - Acomodate the tests to the changes
 - Sync test data with API response
 - Change the repo keys to be in sync with how we use them on the node repo checks
 - Acomodate the controller to deal with the API returning a products view rather than a per-repo response.

Itxaka commented:
![](https://github.trello.services/images/mini-trello-icon.png) [(3) S49P19 #32 Admin Server Repo Checks: Service Integration](https://trello.com/c/oGniN5Lf/149-3-s49p19-32-admin-server-repo-checks-service-integration)

santiph commented on assets/app/features/upgrade/upgrade.constants.js:
```diff
@@ -7,5 +7,9 @@
+        })
+        .constant('ADMIN_REPO_CHECKS_MAP', {
+            'os': ['SLES-12-SP2', 'SLES-12-SP2-Updates'],
+            'openstack': ['SLES-OpenStack-Cloud-7','SLES-OpenStack-Cloud-7-Updates']
```
Could NODES reuse the Admin definition somehow? I'd like to avoid code repetition.

Itxaka replied:
Yup. Should consolidate both of them into just one constant. Incoming.

Itxaka replied:
done!

santiph commented on routes/api/crowbar/repocheck.js:
```diff
@@ -1,14 +1,35 @@
 var express = require('express'),
     router = express.Router();
+var repo_correct = false;
```
Please declare new variables by reusing one single `var`

Itxaka replied:
done!

santiph commented on assets/app/features/upgrade/controllers/upgrade-administration-repositories-checks.controller.js:
```diff
@@ -12,29 +12,31 @@
-                'SLES_12_SP2': {
+                'SLES-12-SP2': {
                     status: false, 
                     label: 'upgrade.steps.admin-repository-checks.repositories.codes.SLES_12_SP2'
```
Please, also update language keys to match the repo names

Itxaka replied:
done!

santiph commented on assets/app/features/upgrade/controllers/upgrade-administration-repositories-checks.controller.js:
```diff
@@ -52,27 +54,22 @@
+                        // Iterate over our map
+                        _.forEach(ADMIN_REPO_CHECKS_MAP, function (repos, type) {
+                            _.forEach(repos, function(repo) {
+                                vm.repoChecks.checks[repo].status = repoChecksResponse.data[type]['available']
```
Is there any difference between `repoChecksResponse.data[type]['available']` and `repoChecksResponse.data[type].available`?

Itxaka replied:
Nope, sorry, that is python style applied to javascript, I'll fix it to use `.available`

Itxaka replied:
done!

santiph commented on assets/app/features/upgrade/controllers/upgrade-administration-repositories-checks.controller.spec.js:
```diff
@@ -159,8 +165,13 @@ describe('Upgrade Flow - Admin Repositories Checks Controller', function () {
                 assert.isObject(controller.repoChecks.checks);
-                _.forEach(partiallyFailingChecksResponse.data, function(value, key) {
-                    expect(controller.repoChecks.checks[key].status).toEqual(value);
+                /* eslint-disable no-undef */
```
Please add the constant to the globals in line 1 instead

Itxaka replied:
done!

Commit:
- Consolidate ADMIN_REPO_CHECKS_MAP and NODE_REPO_CHECKS_MAP into PRODUCTS_REPO_CHECKS_MAP
 - Sync naming for repos keys and admin repo check page
 - Sync label names for repos
 - On admin runRepoChecks only check for os and openstack repos
 - Add PRODUCTS_REPO_CHECKS_MAP to global on spec
 - Use openstack instead of cloud for fake API response
 - Rename NODE_REPO_CHECKS_MAP to PRODUCTS_REPO_CHECKS_MAP in the node repo checks controller+spec

santiph commented on routes/api/crowbar/repocheck.js:
```diff
@@ -1,14 +1,35 @@
+            'repos': {
+                'x86_64': {
+                    'missing': [
+                        'SUSE OpenStack Cloud 7',
```
I think this should be updated to use same repository names listed in 
https://trello.com/c/oGniN5Lf/149-3-s49p19-32-admin-server-repo-checks-service-integration#comment-57f63b68212d25bfb1fd57af

Itxaka replied:
Indeed, the api docs make this clear. I'll change it

Itxaka replied:
Uh, actually the /api/crowbar/repocheck  docs indicate that this is correct this way:

```
{
  "os": {
    "available": true,
    "repos": {}
  },
  "cloud": {
    "available": false,
    "repos": {
      "x86_64": {
        "missing": [
          "SUSE OpenStack Cloud 7"
        ]
      }
    }
  }
}
```

santiph replied:
That's what you should've changed as part of https://trello.com/c/sl8vHEaY/146-3-s49p14-bsc-999946-upgrade-check-for-repositories-on-admin-server-only-checks-for-product

Itxaka committed:
Update fake API response

santiph commented:
Approved
